src/main/java/com/specialwarriors/conal/common/auth/oauth/CustomOAuth2UserService.java

```diff
@@ -0,0 +1,49 @@
+package com.specialwarriors.conal.common.auth.oauth;
+
+import com.specialwarriors.conal.user.domain.User;
+import com.specialwarriors.conal.user.repository.UserRepository;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.oauth2.client.userinfo.DefaultOAuth2UserService;
+import org.springframework.security.oauth2.client.userinfo.OAuth2UserRequest;
+import org.springframework.security.oauth2.client.userinfo.OAuth2UserService;
+import org.springframework.security.oauth2.core.OAuth2AuthenticationException;
+import org.springframework.security.oauth2.core.user.DefaultOAuth2User;
+import org.springframework.security.oauth2.core.user.OAuth2User;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CustomOAuth2UserService implements OAuth2UserService<OAuth2UserRequest, OAuth2User> {
+
+    private final UserRepository userRepository;
+
+    @Override
+    public OAuth2User loadUser(OAuth2UserRequest userRequest) throws OAuth2AuthenticationException {
+        OAuth2UserService<OAuth2UserRequest, OAuth2User> delegate =
+                new DefaultOAuth2UserService();
+        OAuth2User oauth2User = delegate.loadUser(userRequest);
+
+        // GitHub 사용자 정보 추출
+        Number idRaw = oauth2User.getAttribute("id");
+        Long githubId = idRaw.longValue();
+        String username = oauth2User.getAttribute("login");
+        String avatarUrl = oauth2User.getAttribute("avatar_url");
+
+        User user = userRepository.findByGithubId(githubId)
+                .orElseGet(() -> {
+                    return userRepository.save(new User(githubId, username, avatarUrl));
+                });
```

Comment:
```suggestion
User user = userRepository.findByGithubId(githubId)
        .orElseGet(() -> userRepository.save(new User(githubId, username, avatarUrl)));
```
위 코드는 다음과 같이 중괄호를 생략하고 작성할 수 있을 것 같습니다. 한편, 명확성과 가독성을 고려하면 if-else 문으로 나누어 작성할 수도 있을 듯 싶은데 의견이 궁금합니다.

Reply:
중괄호 생략으로 수정했습니다
if-else문을 사용하면 코드 줄 수가 길어져서 오히려 스트림 형식이 깔끔하다고 생각했습니다